src/components/Charts/WindTimeSeries.tsx

```diff
@@ -78,96 +80,112 @@ addWindBarbModule(Highcharts)
  * Wind time series chart component that can display steady and gust speeds
  * in addition to wind barbs for wind direction
  */
-export class WindTimeSeriesChartBase extends React.Component<Props, object> {
-  public render() {
-    const { unit_system } = this.props
-    // Extract wind direction from windspeed data
-    const speeds = this.props.data.filter((d) => !d.name.toLowerCase().includes("direction"))
-    const directions = this.props.data.filter((d) => d.name.toLowerCase().includes("direction"))
-
-    const daysAgo = new Date()
-    daysAgo.setDate(daysAgo.getDate() - this.props.days)
-
-    const filteredSpeeds = speeds.map((d, index) => {
-      const data = d.timeSeries
-        .filter((r) => r.time > daysAgo)
-        .map((r) => [r.time.valueOf(), round(data_converter.convertToNumber(r.reading, unit_system), 1)])
-      return data
-    })
-
-    if (filteredSpeeds[0].length < 1) {
-      return null
-    }
-    const speedsSeries = speeds.map((d, index) => {
-      // Filter windspeeds to only include data from the time range
-      const data = d.timeSeries
+// export class WindTimeSeriesChartBase extends React.Component<Props, object> {
+//   public render() {
+export const WindTimeSeriesChartBase: React.FunctionComponent<Props> = ({
+  unit_system,
+  speed,
+  gust,
+  direction,
+  days,
+  barbsPerDay,
+  height,
+  legend,
+}) => {
+  const speeds: DataTimeSeries[] = []
+  if (speed) {
+    speeds.push(speed)
+  }
+  if (gust) {
+    speeds.push(gust)
+  }
+
+  const speedsSeries = speeds.map((d, index) => {
+    // Filter windspeeds to only include data from the time range
+    let data: number[][] = []
+    if (days) {
+      const daysAgo = new Date()
+      daysAgo.setDate(daysAgo.getDate() - days)
+
+      data = d.timeSeries
         .filter((reading) => reading.time > daysAgo)
         .map(
           // Return Highcharts Spline dataformat [date, reading]
-          (r) => [r.time.valueOf(), round(data_converter.convertToNumber(r.reading, unit_system), 1)]
+          (r) => [new Date(r.time).valueOf(), round(data_converter.convertToNumber(r.reading, unit_system), 1)]
```

Comment:
Codacy found an issue: [Identifier 'data_converter' is not in camel case.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)